identity-wallet/src/state/core_utils/history_event.rs

```diff
@@ -41,3 +41,13 @@ impl HistoryCredential {
         }
     }
 }
+
+/// Implement PartialEq for HistoryEvent to allow for comparison of HistoryEvent instances in tests.
+impl PartialEq for HistoryEvent {
+    fn eq(&self, other: &Self) -> bool {
+        self.connection_id == other.connection_id
+            && self.connection_name == other.connection_name
+            && self.event_type == other.event_type
```

Comment:
This `PartialEq` as well as the `PartialEq` impl for `Connection` are solely for testing use-cases. We currently don't need this implementation elsewhere and I assume it is unlikely that we will need it in the near future.

But it is a bit confusing like this, so eventually it will be cleaner to put these `impl`s behind some "test" feature flag. For now I added a TODO to make it a bit more clear :+1: